hdk-rust/tests/integration_test.rs

```diff
@@ -302,10 +303,67 @@ fn example_valid_entry_address() -> Address {
     Address::from("QmefcRdCAXM2kbgLW2pMzqWhUvKSDvwfFSVkvmwKvBQBHd")
 }
 
+#[allow(dead_code)]
+fn wait_for_action(
+    signal_receiver: &SignalReceiver,
+    expected_signal: ConsistencyEvent,
+    max_tries: i8,
+    up_till: Option<usize>,
+) -> Result<ConsistencyEvent, HolochainError> {
+    if max_tries > 0 {
+        signal_receiver
+            //THIS FIRST PART IS JUST FOR OPTIMIZATION PURPOSES WHILE FINDING A GOOD BALANCE OF SEPERATION
```

Comment:
What do you mean by balance of separation?

Reply:
Seperation of concerns from a high order function perspective but will add that